src/service/chat.rs

```diff
@@ -340,11 +340,140 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn chat_client_delegates_send_message() {
+    async fn test_chat_client_send_message() {
```

Comment:
Let's remove all of these mocked tests.  They don't provide much value to actually testing the code.  Let's only unit test the functionality that has been abstracted out of the client.